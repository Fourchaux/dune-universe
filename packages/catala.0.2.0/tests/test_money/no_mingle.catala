@Article@

/*
new scope A:
  param x content money
  param y content money 
  param z content money

scope A:
  def x := $123.54
  def y := $8548,650.96
  def z := (x *$ y)
*/