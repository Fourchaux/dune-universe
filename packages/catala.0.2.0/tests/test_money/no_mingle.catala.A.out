[ERROR] Error during typechecking, types decimal and money are incompatible
[ERROR] 
[ERROR] Type decimal coming from expression:
[ERROR]   --> test_money/no_mingle.catala
[ERROR]    | 
[ERROR] 12 |   def z := (x *$ y)
[ERROR]    |               ^^
[ERROR] 
[ERROR] Type money coming from expression:
[ERROR]  --> test_money/no_mingle.catala
[ERROR]   | 
[ERROR] 4 | new scope A:
[ERROR]   |           ^
[ERROR] 
[ERROR] Type decimal → money coming from expression:
[ERROR]   --> test_money/no_mingle.catala
[ERROR]    | 
[ERROR] 12 |   def z := (x *$ y)
[ERROR]    |               ^^
[ERROR] 
[ERROR] Type money → any type coming from expression:
[ERROR]   --> test_money/no_mingle.catala
[ERROR]    | 
[ERROR] 12 |   def z := (x *$ y)
[ERROR]    |             ^^^^^^
[ERROR] 
[ERROR] Type money → decimal → money coming from expression:
[ERROR]   --> test_money/no_mingle.catala
[ERROR]    | 
[ERROR] 12 |   def z := (x *$ y)
[ERROR]    |               ^^
[ERROR] 
[ERROR] Type money → money → any type coming from expression:
[ERROR]   --> test_money/no_mingle.catala
[ERROR]    | 
[ERROR] 12 |   def z := (x *$ y)
[ERROR]    |             ^^^^^^
